x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_columns/use_columns.ts

```diff
@@ -162,13 +170,19 @@ export const useColumns = ({
   );
 
   useEffect(() => {
-    // if defaultColumns have changed, populate again
+    // if defaultColumns have changed,
```

Comment:
instead of doing all this work just change condition in the second usedEffect
After:
`if (isEmpty(browserFields) || isColumnsPopulated) return;`
Before:
`if (isBrowserFieldDataLoading !== false || isColumnsPopulated) return;`